### PR TITLE
No re-use of trajectory node proto in SerializeTrajectoryNodes

### PR DESCRIPTION
**Pull request by schwoere**
When serializing `TrajectoryNode`, the declared proto is currently re-used. This can lead to unexpected behavior, in particular when the serialization of the proto is performed in an external function.

This PR also relates PR #1277 where exactly that problem occurred in the SerializeSubmaps.

### CODE REVIEW

**Comment by schwoere** [comment]
I am interested in your opinion about that. It does not seemed to be necessary as long as the `ToProto`-methods always return a proto. The declaration "`SerializedData proto;`" outside the for-loops is also done in `SerializeTrajectoryData`, `SerializeImuData`, `SerializeOdometryData` and `SerializeLandmarkNodes`.

**Comment by schwoere** [comment]
Updated the rest of the serialization methods. ptal @wohe

File: cartographer/io/internal/mapping_state_serialization.cc [hunk 1]
```diff
@@ -99,8 +99,8 @@ void SerializeSubmaps(
 void SerializeTrajectoryNodes(
     const MapById<NodeId, TrajectoryNode>& trajectory_nodes,
     ProtoStreamWriterInterface* const writer) {
-  SerializedData proto;
   for (const auto& node_id_data : trajectory_nodes) {
+    SerializedData proto;
```
[the remark this answers is not in the document]
Comment (schwoere) [reply cartographer/io/internal/mapping_state_serialization.cc]: okay, thanks for the style explanation. I updated the PR.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**